addons/purchase_mrp/models/mrp_production.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+
+
+class MrpProduction(models.Model):
+    _inherit = 'mrp.production'
+
+    purchase_order_count = fields.Integer(
+        "Count of generated PO",
+        compute='_compute_purchase_order_count',
+        groups='purchase.group_purchase_user')
+
+    @api.depends('procurement_group_id.stock_move_ids.created_purchase_line_id.order_id')
+    def _compute_purchase_order_count(self):
+        for production in self:
+            production.purchase_order_count = len(production.procurement_group_id.stock_move_ids.created_purchase_line_id.order_id)
+
+    def _get_document_iterate_key(self, move_raw_id):
+        iterate_key = super(MrpProduction, self)._get_document_iterate_key(move_raw_id)
+        if not iterate_key and move_raw_id.created_purchase_line_id:
+            iterate_key = 'created_purchase_line_id'
+        return iterate_key
```

Comment:
according to our guidelines this method should go at the bottom of the file don't you think?

Private attributes (_name, _description, _inherit, …)
Default method and _default_get
Field declarations
Compute, inverse and search methods in the same order as field declaration
Selection method (methods used to return computed values for selection fields)
Constrains methods (@api.constrains) and onchange methods (@api.onchange)
CRUD methods (ORM overrides)
Action methods
And finally, other business methods.